app/src/pages/dataset/DatasetExamplePage.tsx

```diff
@@ -0,0 +1,170 @@
+import CodeMirror from "@uiw/react-codemirror";
+import { css } from "@emotion/react";
+
+import {
+  Card,
+  CardProps,
+  Dialog,
+  DialogContainer,
+  Flex,
+  View,
+} from "@arizeai/components";
+
+import { CopyToClipboardButton } from "@phoenix/components";
+import { useTheme } from "@phoenix/contexts";
+
+import type { DatasetExamplePageQuery } from "./__generated__/DatasetExamplePageQuery.graphql";
+
+/**
+ * A page that shows the details of a dataset example.
+ */
+export function DatasetExamplePage() {
+  const { datasetId, exampleId } = useParams();
+  const data = useLazyLoadQuery<DatasetExamplePageQuery>(
+    graphql`
+      query DatasetExamplePageQuery($exampleId: GlobalID!) {
+        example: node(id: $exampleId) {
+          ... on DatasetExample {
+            input
+            output
+            metadata
+          }
+        }
+      }
+    `,
+    { exampleId: exampleId as string }
+  );
+  const { input, output, metadata } = useMemo(() => {
+    const example = data.example;
+    return {
+      input: JSON.stringify(example.input),
+      output: JSON.stringify(example.output),
+      metadata: JSON.stringify(example.metadata),
+    };
+  }, [data]);
+  const navigate = useNavigate();
+  return (
+    <DialogContainer
+      type="slideOver"
+      isDismissable
+      onDismiss={() => navigate(`/datasets/${datasetId}`)}
+    >
+      <Dialog size="fullscreen" title={`Example: ${datasetId}`}>
+        <div
+          css={css`
+            overflow-y: auto;
+            padding: var(--ac-global-dimension-size-400);
+          `}
+        >
+          <Flex direction="row" justifyContent="center">
+            <View width="900px" paddingStart="auto" paddingEnd="auto">
+              <Flex direction="column" gap="size-200">
+                <Card
+                  title="Input"
+                  {...defaultCardProps}
+                  extra={<CopyToClipboardButton text={input} />}
+                >
+                  <JSONBlock>{input}</JSONBlock>
+                </Card>
+                <Card
+                  title="Output"
+                  {...defaultCardProps}
+                  extra={<CopyToClipboardButton text={output} />}
+                >
+                  <JSONBlock>{output}</JSONBlock>
+                </Card>
+                <Card
+                  title="Metadata"
+                  {...defaultCardProps}
+                  extra={<CopyToClipboardButton text={metadata} />}
+                >
+                  <JSONBlock>{metadata}</JSONBlock>
+                </Card>
+              </Flex>
+            </View>
+          </Flex>
+        </div>
+      </Dialog>
+    </DialogContainer>
+  );
+}
+
+const defaultCardProps: Partial<CardProps> = {
+  backgroundColor: "light",
+  borderColor: "light",
+  variant: "compact",
+  collapsible: true,
+};
+
+/**
+ * A block of JSON content that is not editable.
+ */
+export function JSONBlock({ children }: { children: string }) {
```

Comment:
I think ultimately I'm probably ready to converge on a single component for all this at this point (https://github.com/Arize-ai/phoenix/blob/datasets/app/src/components/code/JSONEditor.tsx)

But I can do this refactor in a follow-up